invenio_rdm_records/resources/serializers/csl/schema.py

```diff
@@ -0,0 +1,122 @@
+# -*- coding: utf-8 -*-
+#
+# Copyright (C) 2021 CERN.
+# Copyright (C) 2021 Northwestern University.
+#
+# Invenio-RDM-Records is free software; you can redistribute it and/or modify
+# it under the terms of the MIT License; see LICENSE file for more details.
+
+"""CSL based Schema for Invenio RDM Records."""
+
+from edtf import parse_edtf
+from edtf.parser.grammar import ParseException
+from invenio_access.permissions import system_identity
+from invenio_vocabularies.proxies import current_service as vocabulary_service
+from marshmallow import Schema, fields, missing, post_dump
+from marshmallow_utils.fields import SanitizedUnicode
+
+
+class CreatorSchemaCSL(Schema):
+    """Creator/contributor common schema."""
+
+    family = fields.Str(attribute="person_or_org.name")
+
+    @post_dump(pass_many=False)
+    def capitalize_name_type(self, data, **kwargs):
+        """Capitalize type."""
+        if data.get("nameType"):
+            data["nameType"] = data["nameType"].capitalize()
+
+        return data
+
+
+class CSLJSONSchema(Schema):
+    """DataCite 4.3 Marshmallow Schema."""
+
+    publisher = fields.Str(attribute='metadata.publisher')
+    DOI = fields.Method('get_DOI')
+    language = fields.Method('get_language')
+    title = fields.Method('get_title')
+    issued = fields.Method('get_issued')
+    abstract = fields.Method('get_abstract')
+    author = fields.List(
+        fields.Nested(CreatorSchemaCSL), attribute='metadata.creators')
+    note = fields.Method('get_note')
+    version = SanitizedUnicode(attribute="metadata.version")
+    # PIDS-FIXME: What about versioning links and related ids
+    type = fields.Method('get_type')
+    id = fields.Str(attribute='id')
+
+    def get_DOI(self, obj):
+        """Get DOI."""
+        DOI = obj["links"]["self_doi"]
+        if DOI:
+            return DOI
+
+        return missing
+
+    def get_language(self, obj):
+        """Get language."""
+        metadata = obj["metadata"]
+        languages = metadata.get("languages")
+
+        if languages:
+            # PIDS-FIXME: How to choose? the first?
+            return languages[0]["id"]
+
+        return missing
+
+    def get_title(self, obj):
+        """Get title."""
+        metadata = obj["metadata"]
+        return metadata.get("title", missing)
+
+    def get_issued(self, obj):
+        """Get issued dates."""
+        date_parts = []
+        publication_date = obj["metadata"]["publication_date"]
+
+        for splited_date in publication_date.split("/"):
```

Comment:
I think that `-` is accepted as date separator. You probably need to parse it before splitting it.
I propose to merge this and work on the fix on the next one. Don't forget to fix it in the next one.
WDYT @ppanero @Pineirin?

Reply:
I didn't understand that in the documentation, but I can be wrong. Still, I prefer also to merge, as I already updated the other branch and I can implement in it changes of the json serializer without worrying about two separate branches.

Reply:
Also I corrected some unused imports, so it's better for not needing to update both branches.

Reply:
@ntarocco yes, I agree. The `publication_date` is already parsed at record creation time, so whatever string we get here is a EDTF compatible string and we can safely do `parse_edtf(publication_date)` - i.e. without splitting.

I am ok with merging and fixing this in a different PR, just please keep it in separate commits. i.e. one for this fix, one for the actual `CitationStringSerializer`. @Pineirin